### PR TITLE
[Ini/Serialize] Implement logic to serialize

### PR DESCRIPTION
**Pull request by zhoonit**
- [Ini/Serialize] Implement logic to serialize 

```
This patch implements general concept of serializing a node.

Below is the plan to implement this logic

0. Implement logic to serialize [*]
1. Add export api
2. Add exporter
3. Add export test
4. Implement properties
5. combine with export
6. Add extra test

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/compiler/ini_interpreter.cpp [hunk 1]
```diff
@@ -247,7 +249,22 @@ getMergeableGraph(std::shared_ptr<const GraphRepresentation> graph,
 
 void IniGraphInterpreter::serialize(
   std::shared_ptr<const GraphRepresentation> representation,
-  const std::string &out) {}
+  const std::string &out) {
+
+  std::vector<IniSection> sections;
+  for (const auto &ln : representation->getSorted()) {
```
Comment (kparichay) [inline nntrainer/compiler/ini_interpreter.cpp:255]: any reason to use `getSorted()`?

Comment (kparichay) [reply nntrainer/compiler/ini_interpreter.cpp:255]: The ini representation is independent of the order of the sections.

Comment (zhoonit) [reply nntrainer/compiler/ini_interpreter.cpp:255]: AFAIK, there are some bugs if first layer is not input layer or output layer is before input layer and so on..
Even though above is sorted out, it is neater and more human readable to make it sorted.

Comment (kparichay) [reply nntrainer/compiler/ini_interpreter.cpp:255]: > AFAIK, there are some bugs if first layer is not input layer or output layer is before input layer and so on..

I think this was resolved when the network_graph was refactored (not yet with backbone though).

Comment (kparichay) [reply nntrainer/compiler/ini_interpreter.cpp:255]: > Even though above is sorted out, it is neater and more human readable to make it sorted.

Then lets use unsorted representation in debug mode to make sure that the all styles work.

Comment (zhoonit) [reply nntrainer/compiler/ini_interpreter.cpp:255]: Let me have a todo instead, I am also testing backbone.

Comment (kparichay) [reply nntrainer/compiler/ini_interpreter.cpp:255]: ok.

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**